Lib/pydoc.py

```diff
@@ -284,6 +285,9 @@ def visiblename(name, all=None, obj=None):
     # Namedtuples have public fields and methods with a single leading underscore
     if name.startswith('_') and hasattr(obj, '_fields'):
         return True
+    if name in __future__.all_feature_names:
```

Comment:
At the risk of sounding like a pedant, would making this a set outside the function and checking it here be faster (however marginally)?

Reply:
> do we need tests for this?

It is tested - if you take the change I made to Lib/test/pydoc_mod.py to the main branch the test will fail.

> At the risk of sounding like a pedant, would making this a set outside the function and checking it here be faster (however marginally)?

Sure, there's nothing to lose.